File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
# 0과 1을 비교 같으면 popleft 다르면 append(popleft)

from collections import deque

def solution(arr):
    answer = []
    q_a = deque(arr)
    
    for i in range(0, len(arr)-1):
        
        if q_a[0] == q_a[1]:
            q_a.popleft()
        else:
            answer.append(q_a[0])
            q_a.append(q_a.popleft())
            # q_a.append(q_a.popleft())
    answer.append(q_a[0])
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public ArrayList<Integer> solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        int temp = -1;
        for (int i=0; i<arr.length; i++) {
            if (temp != arr[i]) {
                answer.add(arr[i]);
            }
            temp = arr[i];
        }
        return answer;
    }
}